docs/containers/kubernetes-env-vars.md

```diff
@@ -0,0 +1,96 @@
+---
+Area: containers
+ContentId: 8c1ee589-25bb-4c2d-ab8d-7ccfa7b4e282
+PageTitle: Use Kubernetes service environment variables for service to service communication
+DateApproved: 02/12/2021
+MetaDescription: Learn how to use Kubernetes service environment variables with Bridge to Kubernetes to enable service-to-service communication as a non-elevated user in a Kubernetes cluster
+---
+# Use Kubernetes service environment variables
+
+When you communicate with another service in the same Kubernetes cluster, for example with an HTTP request, you typically use the service name in the URL for the request. Bridge to Kubernetes reroutes traffic by modifying the host name resolution to redirect network traffic to its own version of the services. This works in most scenarios, but fails in the case where the Bridge to Kubernetes process has restricted privilege, such as when the request originates from a non-elevated user account or when using VS Code Remote SSH. This is because in order to enable the name resolution for redirected services, Bridge to Kubernetes needs to modify the hosts file, but that is not possible when Bridge to Kubernetes runs from a non-elevated user account. To work around this issue, you can write your code to use the Kubernetes service environment variables instead of a hardcoded service name.
+
+## Kubernetes service environment variables
+
+The following table shows the Kubernetes service environment variables that are available from any service in the cluster, for an example service using the TCP protocol on a port. The *servicename* is the name of the service, converted to uppercase, and with hyphens converted to underscores, so for example, a service named web-api yields an environment variable named WEB_API_SERVICE_HOST.
+
+| Name | Example | Description |
+| - | - | - |
+| *servicename*_SERVICE_HOST | 10.0.0.11 | The name of the service host |
+| *servicename*_SERVICE_PORT | 6379 | The port for the service |
+| *servicename*_PORT | tcp://10.0.0.11:6379 | The Uri with protocol, IP address, and port. |
+| *servicename*\_PORT_*portnumber*_TCP | tcp://10.0.0.11:6379 | The TCP Uri with IP address and port. |
+| *servicename*\_PORT_*portnumber*_TCP_PROTO| tcp | The protocol identifier. |
+| *servicename*\_PORT_*portnumber*_TCP_PORT | 6379 | The port number for TCP. |
+| *servicename*\_PORT_*portnumber*_TCP_ADDR | 10.0.0.11 | The IP address for TCP. |
+
+So if the service is named web-api, the variables are WEB_API_SERVICE_HOST and WEB_API_SERVICE_PORT, and so on. The default environment variables created by Kubernetes are described in the Kubernetes documentation [here](https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables).
+
+## Using the environment variables in code
```

Comment:
Suggest "Environment variables in source code"
Helps avoid confusion with VS "Code".